### PR TITLE
Enhancement/8553 icei woocommerce

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue:

- #8553 

## Relevant technical choices

* Used the `woocommerce_loaded` action to determine whether the WooCommerce plugin is active or not. This is the recommended way to detect that.
* Updated the Plugin class to register the main conversion tracking class only if the appropriate feature flag is enabled for the site.
* Updated the hook that we use to enqueue provider scripts to do that only if either ads or analytics snippet is inserted.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/event-providers/woocommerce.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+	const body = jQuery( 'body' );
+
+	body.on( 'added_to_cart', () => {
+		global.gtag( 'event', 'add_to_cart' );
+	} );
+
+	body.on( 'checkout_place_order_success', function () {
+		global.gtag( 'event', 'purchase' );
+	} );
```
Comment (tofumatt) [inline assets/js/event-providers/woocommerce.js:30]:
```suggestion
	body.on( 'added_to_cart', () => {
		global.gtag( 'event', 'add_to_cart' );
	} );

	body.on( 'checkout_place_order_success', function () {
		global.gtag( 'event', 'purchase' );
	} );
```

Any reason for the different function styles here? Not a big deal, but it seems odd that one is different for seemingly no reason 😅

**Review by tofumatt — COMMENTED** [review]
Looks good, just one (very) minor question. Then we can merge this 👍🏻

File: assets/js/event-providers/woocommerce.js (hunk 1 above)
Comment (eugene-manuilov) [reply assets/js/event-providers/woocommerce.js]: Good catch. Updated to use the arrow function as well.

File: package.json [hunk 2]
```diff
@@ -56,9 +56,7 @@
 	},
 	"husky": {
 		"hooks": {
-			"pre-commit": "lint-staged",
-			"prepare-commit-msg": "php -f bin/prepare-commit-msg.php $HUSKY_GIT_PARAMS",
-			"commit-msg": "php -f bin/check-commit-msg.php $HUSKY_GIT_PARAMS"
+			"pre-commit": "lint-staged"
```
Comment (tofumatt) [inline package.json:59]: I'm guessing this was committed by mistake, I'm going to remove it and then commit 😄

**Review by tofumatt — APPROVED** [review]
Looks good, I think the `package.json` commits are an accident 😅

File: package.json (hunk 2 above)
Comment (eugene-manuilov) [reply package.json]: Oups... yes, that was not planned to be committed :smile: Thanks, @tofumatt.

**Pull request merged**